pywbem_mock/_providerdependentregistry.py

```diff
@@ -0,0 +1,136 @@
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+#
+
+"""
+This module defines the ProviderDependentRegistry class.
+"""
+
+import os
+import six
+from pywbem._utils import _format
+
+__all__ = ['ProviderDependentRegistry']
+
+
+class ProviderDependentRegistry(object):
+    """
+    A registry for provider dependent files in context of a mock script.
+
+    This registry allows registering additional dependent files in context of
```

Comment:
Should we be more complete here to state the reason for register and lookup.  Thus to register dependent files when a mock script is cached and to look up and validate the mock_script file dependencies when the cached mock script is loaded.

Reply:
The possible use for validating a cache in pywbemtools is already described in the mockwbemserver.rst file just before this class is shown, as:
```
The pywbemtools project makes use of this registry for validating whether its
mock cache is up to date w.r.t. additional dependent files a mock script has
used.
```